src/data/polynomial/eval.lean

```diff
@@ -296,6 +296,17 @@ end
 lemma map_injective (hf : function.injective f): function.injective (map f) :=
 λ p q h, ext $ λ m, hf $ by rw [← coeff_map f, ← coeff_map f, h]
 
+
+variables {f}
+
+lemma map_monic_eq_zero_iff (hp : p.monic) : p.map f = 0 ↔ ∀ x, f x = 0 :=
```

Comment:
I'm wondering, might you want to conclude `unique S` instead (it's not a prop, I know)? Somehow `∀ x, f x = 0` feels like a bit of a roundabout way to say that `S` is the zero ring.
But maybe it's the most useful version in practice...

Reply:
`f` is not surjective here, so do you mean `unique (set.range f)`?

Reply:
But if `f` maps `1` to `0`, you can conclude that `S` must be the zero ring, and `f` must have been surjective after all.
Anyway, that kind of stuff can be added later.

Reply:
Ah of course, that is what I was missing.